app/signup/__tests__/page.spec.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import SignUpPage from '../page';
+
+jest.mock('next/navigation', () => ({
+    useRouter: jest.fn()
+}));
+
+//Need to import this useRouter after the jest.mock is in place
```

Comment:
Interesting, I think this may have been what was tripping me up in my tests.
So this is essentially mocking the entire `next/navigation` module and then you are importing that "mocked" version of the `userRouter` function?

Reply:
Yes that is my understanding. Mocking "useRouter" took me forever to figure out. It was a big headache, but I finally got it to work by importing after the jest.mock.